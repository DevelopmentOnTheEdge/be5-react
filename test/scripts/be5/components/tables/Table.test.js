import React          from 'react';
import renderer       from 'react-test-renderer';
import Table          from '../../../../../src/scripts/be5/components/tables/Table';
import TableForm      from '../../../../../src/scripts/be5/components/tables/TableForm';
import FormTable      from '../../../../../src/scripts/be5/components/tables/FormTable';
import TableFormRow   from '../../../../../src/scripts/be5/components/tables/TableFormRow';
import {shallow, mount, render} from 'enzyme';

import $               from 'jquery';
import dt from 'datatables.net';
dt(window, $);

test('test datatables', () => {

  const wrapper = mount( <Table value={json}/> );

});

test('Table', () => {
  const component = renderer.create(
    <Table value={json}/>
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('TableForm', () => {
  const component = renderer.create(
    <TableForm value={json}/>
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('FormTable', () => {
  const component = renderer.create(
    <FormTable value={json}/>
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('TableFormRow', () => {
  const component = renderer.create(
    <TableFormRow value={json}/>
  );

  expect(component.toJSON()).toMatchSnapshot();
});

const json = {
  "data": {
    "attributes": {
      "category": "companies",
      "columns": [
        "Наименование"
      ],
      "hasAggregate": false,
      "layout": {
        "type": "tableForm"
      },
      "length": 1,
      "operations": [
        {
          "clientSide": false,
          "isClientSide": false,
          "name": "Edit",
          "requiresConfirmation": false,
          "title": "Редактировать",
          "visibleWhen": "oneSelected"
        },
        {
          "clientSide": false,
          "isClientSide": false,
          "name": "Insert",
          "requiresConfirmation": false,
          "title": "Добавить",
          "visibleWhen": "always"
        }
      ],
      "page": "Общие сведения",
      "parameters": {},
      "rows": [
        {
          "cells": [
            {
              "content": "adfd",
              "options": {
                "quick": {
                  "visible": "false"
                }
              }
            }
          ],
          "id": "19"
        }
      ],
      "selectable": true,
      "title": "Организации: Общие сведения",
      "totalNumberOfRows": 1
    },
    "type": "table"
  },
  "links": {
    "self": "table/companies/Общие сведения"
  },
  "meta": {
    "_ts_": "1507198487234"
  }
};