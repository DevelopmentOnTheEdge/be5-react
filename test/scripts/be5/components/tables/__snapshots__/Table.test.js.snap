// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormTable 1`] = `
<div
  className="form-table"
>
  <div
    className="document-content"
    id="document-content___table"
  >
    <div>
      <div
        className="table-component companies_All-records"
      >
        <h1
          className="table-component__title"
        >
          Организации: Общие сведения
        </h1>
        <div
          className="operationList"
        >
          <button
            className="btn btn-secondary btn-secondary-old btn-sm"
            disabled={true}
            onClick={[Function]}
          >
            Редактировать
          </button>
           
          <button
            className="btn btn-secondary btn-secondary-old btn-sm"
            disabled={false}
            onClick={[Function]}
          >
            Добавить
          </button>
        </div>
        <div
          className="table-wrap"
        >
          <div
            id="quickColumns"
          >
            <span>
              Другие колонки:
            </span>
            <span>
              <input
                checked={false}
                id="quick0"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="rowIndex"
                htmlFor="quick0"
              >
                Наименование
                 
              </label>
            </span>
          </div>
          <div
            className="row data-table-wrapper"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Table 1`] = `
<div
  className="table-component companies_All-records"
>
  <h1
    className="table-component__title"
  >
    Организации: Общие сведения
  </h1>
  <div
    className="operationList"
  >
    <button
      className="btn btn-secondary btn-secondary-old btn-sm"
      disabled={true}
      onClick={[Function]}
    >
      Редактировать
    </button>
     
    <button
      className="btn btn-secondary btn-secondary-old btn-sm"
      disabled={false}
      onClick={[Function]}
    >
      Добавить
    </button>
  </div>
  <div
    className="table-wrap"
  >
    <div
      id="quickColumns"
    >
      <span>
        Другие колонки:
      </span>
      <span>
        <input
          checked={false}
          id="quick0"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="rowIndex"
          htmlFor="quick0"
        >
          Наименование
           
        </label>
      </span>
    </div>
    <div
      className="row data-table-wrapper"
    />
  </div>
</div>
`;

exports[`Table on MAIN_DOCUMENT 1`] = `
<div
  className="table-component companies_All-records"
>
  <h1
    className="table-component__title"
  >
    Организации: Общие сведения
  </h1>
  <div
    className="operationList"
  >
    <button
      className="btn btn-secondary btn-secondary-old btn-sm"
      disabled={true}
      onClick={[Function]}
    >
      Редактировать
    </button>
     
    <button
      className="btn btn-secondary btn-secondary-old btn-sm"
      disabled={false}
      onClick={[Function]}
    >
      Добавить
    </button>
  </div>
  <div
    className="table-wrap"
  >
    <div
      id="quickColumns"
    >
      <span>
        Другие колонки:
      </span>
      <span>
        <input
          checked={false}
          id="quick0"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="rowIndex"
          htmlFor="quick0"
        >
          Наименование
           
        </label>
      </span>
    </div>
    <div
      className="row data-table-wrapper"
    />
  </div>
</div>
`;

exports[`Table with topForm 1`] = `
<div
  className="table-component users_All-records"
>
  <div
    className="be5-form users_TopFilter form-inline-mini"
  >
    <form
      className="form-inline"
      onSubmit={[Function]}
    >
      <label
        className="mr-sm-2 col-form-label-sm"
      >
        <strong>
          Пользователи: 
        </strong>
      </label>
      <div
        className="property form-group mr-sm-2"
      >
        <label
          className="form-control-label mr-sm-2 col-form-label-sm"
          htmlFor="user_namePropertyInput"
        >
          Логин
        </label>
        <input
          className="property-input form-control form-control-sm"
          id="user_namePropertyInput"
          maxLength="100"
          onBlur={[Function]}
          onChange={[Function]}
          onInput={[Function]}
          onInvalid={[Function]}
          pattern={undefined}
          placeholder={undefined}
          required={false}
          size={undefined}
          type="text"
          value=""
        />
      </div>
      <div
        className="property form-group mr-sm-2 display-none"
      >
        <label
          className="form-control-label mr-sm-2 col-form-label-sm"
          htmlFor="_search_presets_PropertyInput"
        >
          _search_presets_
        </label>
        <input
          className="property-input form-control form-control-sm"
          id="_search_presets_PropertyInput"
          maxLength={undefined}
          onBlur={[Function]}
          onChange={[Function]}
          onInput={[Function]}
          onInvalid={[Function]}
          pattern={undefined}
          placeholder={undefined}
          readOnly="readonly"
          required={false}
          size={undefined}
          type="text"
          value=""
        />
      </div>
      <div
        className="property form-check mr-sm-2 display-none"
      >
        <input
          checked={true}
          className="property-input form-check-input"
          disabled="disabled"
          id="_search_PropertyInput"
          onChange={[Function]}
          required={false}
          size={undefined}
          type="checkbox"
        />
        <label
          className="form-check-label col-form-label-sm"
          htmlFor="_search_PropertyInput"
        >
          _search_
        </label>
      </div>
      <button
        className="btn btn-primary btn-sm"
        disabled={false}
        onClick={[Function]}
        title=""
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
  <h1
    className="table-component__title"
  >
    Пользователи
  </h1>
  <div
    className="operationList"
  />
  <div
    className="table-wrap"
  >
    <div
      className="row data-table-wrapper"
    />
  </div>
</div>
`;

exports[`TableForm 1`] = `
<div
  className="table-form"
>
  <div
    className="document-content"
    id="document-content___table"
  >
    <div>
      <div
        className="table-component companies_All-records"
      >
        <h1
          className="table-component__title"
        >
          Организации: Общие сведения
        </h1>
        <div
          className="operationList"
        >
          <button
            className="btn btn-secondary btn-secondary-old btn-sm"
            disabled={true}
            onClick={[Function]}
          >
            Редактировать
          </button>
           
          <button
            className="btn btn-secondary btn-secondary-old btn-sm"
            disabled={false}
            onClick={[Function]}
          >
            Добавить
          </button>
        </div>
        <div
          className="table-wrap"
        >
          <div
            id="quickColumns"
          >
            <span>
              Другие колонки:
            </span>
            <span>
              <input
                checked={false}
                id="quick0"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="rowIndex"
                htmlFor="quick0"
              >
                Наименование
                 
              </label>
            </span>
          </div>
          <div
            className="row data-table-wrapper"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`TableFormRow 1`] = `
<div
  className="row"
>
  <div
    className="col-lg-6"
  />
  <div
    className="col-lg-6"
  >
    <div
      className="document-content"
      id="document-content___table"
    >
      <div>
        <div
          className="table-component companies_All-records"
        >
          <h1
            className="table-component__title"
          >
            Организации: Общие сведения
          </h1>
          <div
            className="operationList"
          >
            <button
              className="btn btn-secondary btn-secondary-old btn-sm"
              disabled={true}
              onClick={[Function]}
            >
              Редактировать
            </button>
             
            <button
              className="btn btn-secondary btn-secondary-old btn-sm"
              disabled={false}
              onClick={[Function]}
            >
              Добавить
            </button>
          </div>
          <div
            className="table-wrap"
          >
            <div
              id="quickColumns"
            >
              <span>
                Другие колонки:
              </span>
              <span>
                <input
                  checked={false}
                  id="quick0"
                  onChange={[Function]}
                  type="checkbox"
                />
                <label
                  className="rowIndex"
                  htmlFor="quick0"
                >
                  Наименование
                   
                </label>
              </span>
            </div>
            <div
              className="row data-table-wrapper"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`noRecordsOnThePage 1`] = `
<div
  className="table-component companies_All-records"
>
  <h1
    className="table-component__title"
  >
    Организации: Общие сведения
  </h1>
  <div>
    <p>
      No records on page 11
    </p>
    <ul
      className="pagination"
    >
      <li
        className="paginate_button page-item"
      >
        <a
          className="page-link"
          href="#"
          onClick={[Function]}
        >
          Previous
        </a>
      </li>
      <li
        className="paginate_button page-item"
      >
        <a
          className="page-link"
          href="#"
          onClick={[Function]}
        >
          1
        </a>
      </li>
      <li
        className="paginate_button page-item disabled"
      >
        <a
          className="page-link"
          href="#"
          onClick={[Function]}
        >
          ...
        </a>
      </li>
      <li
        className="paginate_button page-item"
      >
        <a
          className="page-link"
          href="#"
          onClick={[Function]}
        >
          10
        </a>
      </li>
      <li
        className="paginate_button page-item disabled"
      >
        <a
          className="page-link"
          href="#"
        >
          Next
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`tableWithFilterInfo 1`] = `
<div
  className="table-component companies_All-records"
>
  <h1
    className="table-component__title"
  >
    Организации: Общие сведения
     
    <small>
      for Name: Test name
    </small>
  </h1>
  <div
    className="table-wrap"
  >
    <div
      className="row data-table-wrapper"
    />
  </div>
</div>
`;
